### PR TITLE
fix: make deploy target

### PR DESCRIPTION
**Pull request by vprashar2929**
This PR address the issue #381 by:

* Enabling cert-manager that is necessary by webhooks during the
      deployment of Operator on k8s.

* Added a separate config for k8s that contains the manifests that
   are specific for deploying on k8s.
   * `config/default` points to OpenShift specific manifests
   * `config/k8s` points to manifests specific for k8s
    
* Added overlays in manager to isolate k8s and
      OpenShift

### CODE REVIEW

File: bundle/manifests/kepler.system.sustainable.computing.io_keplers.yaml [hunk 1]
```diff
@@ -2,6 +2,7 @@ apiVersion: apiextensions.k8s.io/v1
 kind: CustomResourceDefinition
 metadata:
   annotations:
+    cert-manager.io/inject-ca-from: kepler-operator-system/kepler-operator-serving-cert
```
Comment (sthaha) [inline bundle/manifests/kepler.system.sustainable.computing.io_keplers.yaml:5]: How does this behave on OpenShift that has cert-manager installed?

File: Makefile [hunk 2]
```diff
@@ -245,7 +248,10 @@ uninstall: manifests kustomize ## Uninstall CRDs from the K8s cluster specified
 		kubectl delete --ignore-not-found=$(ignore-not-found) -f -
 
 .PHONY: deploy
-deploy: install ## Deploy controller to the K8s cluster specified in ~/.kube/config.
+deploy: install cmctl ## Deploy controller to the K8s cluster specified in ~/.kube/config.
+	kubectl apply --server-side --force-conflicts -f \
+		https://github.com/jetstack/cert-manager/releases/download/v$(CERTMANAGER_VERSION)/cert-manager.yaml
```
Comment (sthaha) [inline Makefile:253]: Imagine running this on k8s with a cert-manager already installed and configured? Can this potentially break their system? 

IMO, we should first validate if certmanager exists and install it only if there isn't any. 
Additionally, consider exiting with error if cert-manager is absent. 
We can also provide `install-certmanager` helper to make life easier for vanilla k8s installation.

File: hack/cluster.sh [hunk 3]
```diff
@@ -31,7 +31,6 @@ declare -r PROJECT_ROOT
 declare -r TMP_DIR="$PROJECT_ROOT/tmp"
 declare -r DEV_CLUSTER_DIR="$TMP_DIR/local-dev-cluster"
 declare -r BIN_DIR="$TMP_DIR/bin"
-declare -r OPERATOR_SDK_VERSION=${OPERATOR_SDK_VERSION:-v1.27.0}
```
Comment (sthaha) [inline hack/cluster.sh:34]: any reason to remove this?

**Review by sthaha — COMMENTED** [review]
Thanks a lot @vprashar2929 !

File: hack/cluster.sh (hunk 3 above)
Comment (vprashar2929) [reply hack/cluster.sh:34]: wasn't used in the script anywhere

File: config/default/manager_config_patch.yaml
Comment (vprashar2929) [inline config/default/manager_config_patch.yaml:1]: Deleted this file as it wasn't used anywhere

**Pull request closed without merging**